.github/workflows/integration.yml

```diff
@@ -97,14 +82,20 @@ jobs:
             ~/.cargo/git/db/
             target/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-      # Run an initial build in a separate step to split the build time from execution time
+      # Run an initial build in a separate step to split the build time from execution time    
       - name: Build bins
         run: cargo build --bin gen_blockchain_data
       - name: Build tests
         run: for testname in rpc circuit_input_builder circuits; do cargo test --profile release --test $testname --features $testname --no-run; done
       - run: ./run.sh --steps "setup"
+        working-directory: integration-tests
```

Comment:
I noticed that this directive was previously specified in "defaults". I think it should also be applied for "Build bins" and "Build tests".  Maybe it also works from the root of the repository, but I think the "Build tests" step may build much more than necessary (so it will take more time than necessary)

Reply:
working-directory was removed from the default section, because during the workflow execution we received a "No such file or directory" error message. With current workflows a generic image is used to deploy the runner, via github-ops, therefore at this specific execution point, the expected dir structure is not there and the default/working-directory fails. As a result, we need to set working directory after checkout step, for each separate run directive of the job. 

"Build bins" and "Build tests" work fine from the default working repo directory. Of course if there is a reason that we 're missing we can adapt.